packages/grpc-js/src/http_proxy.ts

```diff
@@ -119,6 +119,63 @@ function getNoProxyHostList(): string[] {
   }
 }
 
+interface CIDRNotation {
+  ip: number;
+  prefixLength: number;
+}
+
+/*
+ * The groups correspond to CIDR parts as follows:
+ * 1. ip
+ * 2. prefixLength
+ */
+
+export function parseCIDR(cidrString: string): CIDRNotation | null {
+  const splitRange = cidrString.split('/');  
+  if (splitRange.length !== 2) {  
+    return null;  
+  }  
+  const prefixLength = parseInt(splitRange[1], 10);  
+  if (!isIPv4(splitRange[0]) || Number.isNaN(prefixLength) || prefixLength < 0 || prefixLength > 32) {  
+    return null;  
+  }  
+  return {  
+    ip: ipToInt(splitRange[0]),  
+    prefixLength: prefixLength  
+  };
+}
+
+function ipToInt(ip: string) {
+  return ip.split(".").reduce((acc, octet) => (acc << 8) + parseInt(octet, 10), 0);
+}
+
+function isIpInCIDR(cidr: CIDRNotation, serverHost: string) {
+  const ip = cidr.ip;
+  const mask = -1 << (32 - cidr.prefixLength);
+  const hostIP = ipToInt(serverHost);
+
+  return (hostIP & mask) === (ip & mask);
+}
+
+function hostMatchesNoProxyList(serverHost: string): boolean {
+  for (const host of getNoProxyHostList()) {
+    const parsedCIDR = parseCIDR(host);
+    // host is a single IP or a domain name suffix
+    if (!parsedCIDR) {
```

Comment:
`if...else` blocks should not use negated conditions. Please switch the two cases.